src/main/java/uk/gov/pay/api/resources/PaymentsResource.java

```diff
@@ -105,15 +107,16 @@ public PaymentsResource(Client client,
     public Response getPayment(@ApiParam(value = "accountId", hidden = true) @Auth Account account,
                                @PathParam("paymentId")
                                @ApiParam(name = "paymentId", value = "Payment identifier", example = "hu20sqlact5260q2nanm0q8u93")
-                                       String paymentId) {
+                                       String paymentId,
+                               @HeaderParam("X-Ledger") String strategyName) {
```

Comment:
maybe deserialise this to an enum, then validation can be done there, and the strategy can just `execute`?

Reply:
But then it will throw and I wanted it to just log and fallback to the default strategy...
But maybe it's not a good idea?

Reply:
having discussed, this is temporary, so no need to try and do something over fancy. LGTM!